lib/src/useStep/useStep.ts

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useMemo, useState } from 'react'
+
+interface ReturnType {
+  actualStep: number
+  goToNextStep: () => void
+  goToPrevStep: () => void
+  reset: () => void
+  canGoToNextStep: boolean
+  canGoToPrevStep: boolean
+  setStep: (step: number) => void
+}
```

Comment:
I think returning a tuple is more conventional for a hook, like this: 

```suggestion
interface Methods {
  goToNextStep: () => void
  goToPrevStep: () => void
  reset: () => void
  canGoToNextStep: boolean
  canGoToPrevStep: boolean
  setStep: (step: number) => void
}

type ReturnType = [number, Methods]
```